southworkscom/AI
# [Enterprise Bot Template][TS] Update botbuilder dependencies

## Description
Update botbuilder dependencies on the package.json to the latest version:

	• Botbuilder 4.0.6 to 4.2.0
	• Botbuilder-ai 4.0.6 to 4.2.0
	• botbuilder-azure: 4.0.6 to 4.2.0
	• botbuilder-core": 4.0.6 to 4.2.0
	• botbuilder-dialogs: 4.0.6 to 4.2.0
Also, we add the `ms-rest-azure: 2.5.0` because the dependency is needed to build the project.  

Fix the issues that appears when update the botbuilder dependencies:

Delete the argument of type undefined.
![image](https://user-images.githubusercontent.com/42946572/50857205-b3c56f80-136b-11e9-98ca-684783a06a0c.png)

Rename options attribute in TelemetryQnAMaker to "_qnaOptions".
![image](https://user-images.githubusercontent.com/42946572/50857245-d061a780-136b-11e9-9769-9d25465c5d24.png)

Add the properties that missing from Activity like localTimezone, listenFor, semanticAction.
![image](https://user-images.githubusercontent.com/42946572/50857287-ea9b8580-136b-11e9-861e-93c9ae0d2d79.png)

## Related Issue
[Enterprise Bot Template][TypeScript] Problem building bot [#522](https://github.com/Microsoft/AI/issues/522)

## Testing Steps

1. Open a new terminal in AI/templates/Enterprise-Template/src/typescript.
2. Run npm install.
3. Check there is no exceptions.
4. Run npm run build.
5. Check there is no exceptions encountered.

## Checklist

If this contains changes that needs to be replicated between the Enterprise Template <-> Virtual Assistant

If you have updated responses or `.lu` files:
